edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/LogsRequest.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Requests
+{
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.Logs;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Json;
+    using Newtonsoft.Json;
+
+    public class LogsRequest
+    {
+        public LogsRequest(
+            string schemaVersion,
+            List<LogRequestItem> items,
+            LogsContentEncoding encoding,
+            LogsContentType contentType)
+        {
+            this.SchemaVersion = Preconditions.CheckNonWhiteSpace(schemaVersion, nameof(schemaVersion));
+            this.Encoding = encoding;
```

Comment:
Since you're passing default values for null `encoding` and `contentType` values in the `JsonConstructor`, should you `CheckNotNull` here?

Reply:
No encoding and contentType are enums and so cannot be null. 
the JsonCtor takes nullable values and converts them to non-nullable enum values.